OpenID

lelemm opened the pull request:
Cleaning up the PR #381

lelemm commented:
@MatissJanis so, I changed all the review points you pointed out on the old PR, but as a side effect, some stuff needs to change on the UI side. I changed almost everything, but I will need to refactor the user access page.
I will take a look at it tomorrow and then I tell you to review again, ok?

lelemm commented:
You can review again @MatissJanis

lelemm commented on src/app-secrets.js:
```diff
@@ -1,21 +1,38 @@
+
   const { name, value } = req.body;
 
+  if (method === 'openid') {
+    let canSaveSecrets = isAdmin(res.locals.user_id);
+
+    if (!canSaveSecrets) {
+      res.status(403).send({
+        status: 'error',
+        reason: 'not-admin',
+        details: 'You have to be admin to set secrets',
+      });
+
+      return null;
+    }
+  }
```
removing the return would let the code keep executing. I guess I could remove the "null" from the return, but not the return itself

lelemm replied:
done

lelemm commented on src/scripts/reset-password.js:
```diff
@@ -1,13 +1,14 @@
 
-  promptPassword().then((password) => {
-    let { error } = bootstrap(password);
+  promptPassword().then(async (password) => {
+    let { error } = await bootstrap({ password });
```
done

lelemm commented on src/app-admin.test.js:
```diff
@@ -0,0 +1,380 @@
+    describe('GET /users', () => {
+      let sessionUserId, testUserId, sessionToken;
+
+      beforeEach(() => {
+        sessionUserId = uuidv4();
+        testUserId = uuidv4();
+        sessionToken = generateSessionToken();
+
+        createUser(sessionUserId, 'sessionUser', ADMIN_ROLE);
+        createSession(sessionUserId, sessionToken);
+        createUser(testUserId, 'testUser', ADMIN_ROLE);
+      });
+
+      afterEach(() => {
+        deleteUser(sessionUserId);
+        deleteUser(testUserId);
+      });
+
```
can you create a Committable suggestion for this?

lelemm replied:
as said in another point, database calls are not async

lelemm replied:
ok

lelemm commented on src/app-admin.test.js:
```diff
@@ -0,0 +1,380 @@
+const ADMIN_ROLE = 'ADMIN';
+const BASIC_ROLE = 'BASIC';
+
+// Create user helper function
+const createUser = (userId, userName, role, owner = 0, enabled = 1) => {
+  getAccountDb().mutate(
+    'INSERT INTO users (id, user_name, display_name, enabled, owner, role) VALUES (?, ?, ?, ?, ?, ?)',
+    [userId, userName, `${userName} display`, enabled, owner, role],
+  );
+};
+
+const deleteUser = (userId) => {
+  getAccountDb().mutate('DELETE FROM user_access WHERE user_id = ?', [userId]);
+  getAccountDb().mutate('DELETE FROM users WHERE id = ?', [userId]);
+};
+
+const createSession = (userId, sessionToken) => {
+  getAccountDb().mutate(
+    'INSERT INTO sessions (token, user_id, expires_at) VALUES (?, ?, ?)',
+    [sessionToken, userId, Date.now() + 1000 * 60 * 60], // Expire in 1 hour
+  );
+};
+
```
all calls from getAccountDb are NOT async, what is the reasoning behind this?

lelemm replied:
ok

lelemm commented on src/app-admin.js:
```diff
@@ -0,0 +1,394 @@
+let app = express();
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
+app.use(errorMiddleware);
```
done

lelemm commented on migrations/1718889148000-openid.js:
```diff
@@ -0,0 +1,41 @@
+        display_name TEXT,
+        extra_data TEXT, active INTEGER);
+
+    INSERT INTO auth_new (method, display_name, extra_data, active)
+      SELECT 'password', 'Password', password, 1 FROM auth;
```
the value is already encrypted, just moving from one structure to another

lelemm commented on src/accounts/openid.js:
```diff
@@ -0,0 +1,246 @@
+      redirect_uri: client.redirect_uris[0],
+    });
+    const userInfo = await client.userinfo(grant);
+    const identity =
+      userInfo.preferred_username ??
+      userInfo.login ??
+      userInfo.email ??
+      userInfo.id ??
+      userInfo.name;
```
like?

lelemm commented on src/accounts/openid.js:
```diff
@@ -0,0 +1,246 @@
+      [''],
+    );
+    let userId = null;
+    if (countUsersWithUserName === 0) {
+      userId = uuid.v4();
+      accountDb.mutate(
+        'INSERT INTO users (id, user_name, display_name, enabled, owner, role) VALUES (?, ?, ?, 1, 1, ?)',
+        [
+          userId,
+          identity,
+          userInfo.name ?? userInfo.email ?? identity,
+          'ADMIN',
+        ],
+      );
```
This is the expected flow. If the server is compromised before hand, the attacker can already attack

lelemm commented on src/accounts/openid.js:
```diff
@@ -0,0 +1,246 @@
+
+    clearExpiredSessions();
+
+    return { url: `${return_url}/openid-cb?token=${token}` };
```
I'm open for a commitable suggestion

lelemm commented on src/accounts/password.js:
```diff
@@ -0,0 +1,120 @@
+  );
+  let userId = null;
+  if (totalOfUsers === 0) {
+    userId = uuid.v4();
+    accountDb.mutate(
+      'INSERT INTO users (id, user_name, display_name, enabled, owner, role) VALUES (?, ?, ?, 1, 1, ?)',
+      [userId, '', '', 'ADMIN'],
+    );
+  } else {
```
this is how the system works today without openid. without openid there is no multuser, doesn't make a difference to have a name and username, they are never used or displayed

lelemm commented on src/accounts/password.js:
```diff
@@ -0,0 +1,120 @@
+      [userId, '', '', 'ADMIN'],
+    );
+  } else {
+    let { id: userIdFromDb } = accountDb.first(
+      'SELECT id FROM users WHERE user_name = ?',
+      [''],
+    );
+
+    userId = userIdFromDb;
+
+    if (!userId) {
+      return { error: 'user-not-found' };
+    }
+  }
```
same as last comment

lelemm commented on src/accounts/password.js:
```diff
@@ -0,0 +1,120 @@
+  return { token };
+}
+
+export function changePassword(newPassword) {
+  let accountDb = getAccountDb();
+
+  if (!isValidPassword(newPassword)) {
+    return { error: 'invalid-password' };
+  }
+
+  let hashed = hashPassword(newPassword);
+  accountDb.mutate("UPDATE auth SET extra_data = ? WHERE method = 'password'", [
+    hashed,
+  ]);
+  return {};
+}
```
this is done in the expressjs route

lelemm commented on src/app-account.js:
```diff
@@ -26,22 +31,30 @@ export { app as handlers };
 
-app.post('/bootstrap', (req, res) => {
-  let { error, token } = bootstrap(req.body.password);
+app.post('/bootstrap', async (req, res) => {
+  let { error } = await bootstrap(req.body);
```
the bootstrap() method is saniting it, isn't?

lelemm commented on src/accounts/password.js:
```diff
@@ -0,0 +1,120 @@
+  let accountDb = getAccountDb();
+  accountDb.transaction(() => {
+    accountDb.mutate('DELETE FROM auth WHERE method = ?', ['password']);
+    accountDb.mutate('UPDATE auth SET active = 0');
```
no, the update without where is the expected change. when we bootstrap with password, that will be the only enabled auth method, so we disable all the rest

lelemm commented on src/app-openid.js:
```diff
@@ -0,0 +1,91 @@
+export { app as handlers };
+
+app.post('/enable', validateSessionMiddleware, async (req, res) => {
+  if (!isAdmin(res.locals.user_id)) {
```
all database calls are sync when using better-sqlite3, so, there is no async calls needed

lelemm commented on src/app-openid.js:
```diff
@@ -0,0 +1,91 @@
+app.get('/callback', async (req, res) => {
+  let { error, url } = await loginWithOpenIdFinalize(req.query);
+  if (error) {
+    res.status(400).send({ error });
```
add a console log of the message if the error is not being send to the client

lelemm commented on src/app-admin.js:
```diff
@@ -0,0 +1,394 @@
+    return;
+  }
+
+  const ids = req.body.ids;
+  let totalDeleted = 0;
+  ids.forEach((item) => {
+    const ownerId = UserService.getOwnerId();
+
+    if (item === ownerId) return;
+
+    UserService.deleteUserAccess(item);
+    UserService.transferAllFilesFromUser(ownerId, item);
+    const usersDeleted = UserService.deleteUser(item);
+    totalDeleted += usersDeleted;
+  });
```
none of the calls are async

lelemm commented on src/app-account.js:
```diff
@@ -80,22 +102,34 @@ app.post('/login', (req, res) => {
-  let user = validateUser(req, res);
-  if (user) {
-    res.send({ status: 'ok', data: { validated: true } });
+  let session = validateSession(req, res);
+  if (session) {
+    const user = getUserInfo(session.user_id);
+
+    res.send({
+      status: 'ok',
+      data: {
+        validated: true,
+        userName: user?.user_name,
+        permission: user?.role,
+        userId: session?.user_id,
+        displayName: user?.display_name,
+        loginMethod: session?.auth_method,
+      },
+    });
   }
```
its needed

lelemm commented on src/app-openid.js:
```diff
@@ -0,0 +1,91 @@
+});
+
+app.get('/callback', async (req, res) => {
+  let { error, url } = await loginWithOpenIdFinalize(req.query);
```
give example of how to implement isValidRedirectUrl, at the table 'auth', when method = 'openid', extra_data has a server_hostname that is the current domain of the application

lelemm replied:
not sure if this will work:
```
if (redirectUrl.hostname === serverHostname) {
```
because the serverhostname must be (protocol)://hostname (example: https://google.com)

lelemm commented on src/app-openid.js:
```diff
@@ -0,0 +1,91 @@
+    return;
+  }
+
+  let { error } = (await enableOpenID(req.body)) || {};
```
the openid lib validates it and there is a basic validation at enableOpenID

lelemm commented on src/app-admin.js:
```diff
@@ -0,0 +1,393 @@
+    res.status(400).send({
+      status: 'error',
+      reason: 'file-denied',
+      details: "You don't have permissions over this file",
+    });
+    return;
+  }
+
+  const fileIdInDb = UserService.getFileById(userAccess.fileId);
+  if (!fileIdInDb) {
+    res.status(400).send({
+      status: 'error',
+      reason: 'invalid-file-id',
+      details: 'File not found at server',
+    });
+    return;
+  }
+
+  if (!userAccess.userId) {
+    res.status(400).send({
+      status: 'error',
+      reason: 'user-cant-be-empty',
+      details: 'User cannot be empty',
+    });
+    return;
+  }
+
+  if (UserService.countUserAccess(userAccess.fileId, userAccess.userId) > 0) {
+    res.status(400).send({
+      status: 'error',
+      reason: 'user-already-have-access',
+      details: 'User already have access',
+    });
+    return;
+  }
+
+  UserService.addUserAccess(userAccess.userId, userAccess.fileId);
+
+  res.status(200).send({ status: 'ok', data: {} });
+});
```
can't, some routes can't use this as they can accept anonymous

lelemm commented on src/app-sync.js:
```diff
@@ -113,6 +113,8 @@ app.post('/sync', async (req, res) => {
 });
 
 app.post('/user-get-key', (req, res) => {
+  if (!res.locals) return;
+
```
removed this if

lelemm commented on src/load-config.js:
```diff
@@ -77,6 +76,9 @@ let defaultConfig = {
     fileSizeLimitMB: 20,
   },
   projectRoot,
+  multiuser: false,
+  token_expiration: 'never',
+  loginMethod: 'password',
```
that's all true, but currenctly the app supports non expiring token, so I was asked to maintain it

lelemm commented on src/app-sync/services/files-service.js:
```diff
@@ -110,14 +114,53 @@ class FilesService {
+    ).map(this.validate);
+  }
+
+  findUsersWithAccess(fileId) {
+    const userAccess =
+      this.accountDb.all(
+        `SELECT UA.user_id as userId, users.display_name displayName, users.user_name userName
+              FROM files
+                JOIN user_access UA ON UA.file_id = files.id
+                JOIN users on users.id = UA.user_id
+              WHERE files.id = ? 
+          UNION ALL
+        SELECT users.id, users.display_name, users.user_name
+              FROM files
+                JOIN users on users.id = files.owner
+              WHERE files.id = ?
+          `,
+        [fileId, fileId],
+      ) || [];
+
+    return userAccess;
```
this isn't needed since this will be always called after a file find

lelemm commented on src/app-sync/services/files-service.js:
```diff
@@ -99,7 +103,7 @@ class FilesService {
     const deletedInt = boolToInt(file.deleted);
     this.accountDb.mutate(
-      'INSERT INTO files (id, group_id, sync_version, name, encrypt_meta, encrypt_salt, encrypt_test, encrypt_keyid, deleted) VALUES (?, ?, ?, ?, ?, ?, ?,? ,?)',
+      'INSERT INTO files (id, group_id, sync_version, name, encrypt_meta, encrypt_salt, encrypt_test, encrypt_keyid, deleted, owner) VALUES (?, ?, ?, ?, ?, ?, ?, ? ,?, ?)',
```
fixed

lelemm commented on jest.global-setup.js:
```diff
@@ -1,10 +1,80 @@
   await runMigrations();
 
+  createUser(GENERIC_ADMIN_ID, 'admin', ADMIN_ROLE_ID, 1);
+
   // Insert a fake "valid-token" fixture that can be reused
   const db = getAccountDb();
-  await db.mutate('INSERT INTO sessions (token) VALUES (?)', ['valid-token']);
+  await db.mutate('DELETE FROM sessions');
+  await db.mutate(
+    'INSERT INTO sessions (token, expires_at, user_id) VALUES (?, ?, ?)',
+    ['valid-token', -1, 'genericAdmin'],
+  );
+  await db.mutate(
+    'INSERT INTO sessions (token, expires_at, user_id) VALUES (?, ?, ?)',
+    ['valid-token-admin', -1, 'genericAdmin'],
+  );
+
+  setSessionUser('genericAdmin');
+  setSessionUser('genericAdmin', 'valid-token-admin');
```
done

lelemm commented on src/account-db.js:
```diff
@@ -38,74 +45,150 @@ export function getLoginMethod(req) {
-  accountDb.mutate('UPDATE auth SET password = ?', [hashed]);
-  accountDb.mutate('UPDATE sessions SET token = ?', [token]);
+    if (!loginSettings?.password) {
+      return { error: 'invalid-password' };
+    }
 
-  return {};
+    if (passwordHash) {
+      let confirmed = bcrypt.compareSync(loginSettings.password, passwordHash);
+
+      if (!confirmed) {
+        return { error: 'invalid-password' };
+      }
+    }
```
done

lelemm commented on src/accounts/openid.js:
```diff
@@ -0,0 +1,295 @@
+      }
+
+      userId = userIdFromDb;
+    }
```
already done

lelemm commented on src/accounts/openid.js:
```diff
@@ -0,0 +1,296 @@
+  const client = new issuer.Client({
+    client_id: config.client_id,
+    client_secret: config.client_secret,
+    redirect_uri: new URL(
+      '/openid/callback',
+      config.server_hostname,
+    ).toString(),
```
it can accept http too

lelemm commented on src/app-secrets.js:
```diff
@@ -1,21 +1,47 @@
+  }
   const { name, value } = req.body;
 
+  if (method === 'openid') {
+    let canSaveSecrets = isAdmin(res.locals.user_id);
+
+    if (!canSaveSecrets) {
+      res.status(403).send({
+        status: 'error',
+        reason: 'not-admin',
+        details: 'You have to be admin to set secrets',
+      });
+
+      return;
+    }
+  }
```
not changing to this. too confuse

lelemm commented on src/accounts/openid.js:
```diff
@@ -0,0 +1,295 @@
+        'SELECT count(*) as countUsersWithUserName FROM users WHERE user_name <> ?',
+        [''],
+      );
+      if (countUsersWithUserName === 0) {
+        userId = uuid.v4();
+        accountDb.mutate(
+          'INSERT INTO users (id, user_name, display_name, enabled, owner, role) VALUES (?, ?, ?, 1, 1, ?)',
+          [
+            userId,
+            identity,
+            userInfo.name ?? userInfo.email ?? identity,
+            'ADMIN',
+          ],
+        );
+
+        const userFromPasswordMethod = getUserByUsername('');
+        if (userFromPasswordMethod) {
+          transferAllFilesFromUser(userId, userFromPasswordMethod.user_id);
+        }
+      } else {
+        let { id: userIdFromDb, display_name: displayName } =
+          accountDb.first(
+            'SELECT id, display_name FROM users WHERE user_name = ? and enabled = 1',
+            [identity],
+          ) || {};
+
+        if (userIdFromDb == null) {
+          return { error: 'openid-grant-failed' };
+        }
+
+        if (!displayName && userInfo.name) {
+          accountDb.mutate('UPDATE users set display_name = ? WHERE id = ?', [
+            userInfo.name,
+            userIdFromDb,
+          ]);
+        }
+
+        userId = userIdFromDb;
+      }
+    });
```
this return will not work since its inside a anonymous method so the return will be lost

MatissJanis reviewed (approved):
LGTM on my end. Just need another maintainer to review & approve

joel-jeremy reviewed (approved):
LGTM

jfdoming commented on src/app-sync/tests/services/files-service.test.js:
```diff
@@ -156,7 +157,7 @@ describe('FilesService', () => {
 
     // Limit the number of files returned
-    const limitedFiles = filesService.find(1);
+    const limitedFiles = filesService.find({ limit: 1 });
```
I think I agree with the bot here, mind adding some tests for this?

lelemm replied:
added a test for this

jfdoming commented on src/load-config.js:
```diff
@@ -53,7 +53,6 @@ if (process.env.ACTUAL_CONFIG_PATH) {
 
 /** @type {Omit<import('./config-types.js').Config, 'mode' | 'dataDir' | 'serverFiles' | 'userFiles'>} */
 let defaultConfig = {
-  loginMethod: 'password',
```
Curious why this was removed?

lelemm replied:
it was removed (I don't remember why), but I added it back on the end of the defaultConfig object. I moved it back to the top

jfdoming commented on src/app-sync/services/files-service.js:
```diff
@@ -110,14 +116,53 @@ class FilesService {
-      .all('SELECT * FROM files WHERE deleted = 0 LIMIT ?', [limit])
-      .map(this.validate);
+  find({ userId = null, limit = 1000 } = {}) {
+    const canSeeAll = userId === null || isAdmin(userId);
```
This code as written seems a little dangerous to me. In particular, it's used in `app-sync` which just passes in `res.locals.user_id`—right now, it may be the case that this is always non-null, but I can imagine a future PR which accidentally makes that not the case. Do we need the `null` fallback to admin for something? If not, can we remove it and error loudly if the user ID is null?

lelemm replied:
Its nullable because it was messing with the test cases. Fixed the test cases and removed the null

jfdoming commented on src/app-sync.js:
```diff
@@ -314,6 +321,13 @@ app.get('/list-user-files', (req, res) => {
       name: row.name,
       encryptKeyId: row.encryptKeyId,
+      owner: row.owner,
+      usersWithAccess: fileService
+        .findUsersWithAccess(row.id)
+        .map((access) => ({
+          ...access,
+          owner: access.userId === row.owner,
+        })),
```
It may not be for this PR, but this seems a little inefficient to me. I think it would be better to have `findUsersWithAccess(rows.map(({id}) => id))` so we don't have to look at disk for each separate file. What do you think?

(same comment below for `/get-user-file-info`)

jfdoming replied:
Yeah, it makes sense. I'll make a follow-up issue to track this as an optimization

jfdoming reviewed (changes requested):
Left only a couple of minor comments, thanks for pushing through this! I'm requesting changes only so we don't accidentally merge this before we can discuss the comments, but it does look broadly good to me 💯

jfdoming reviewed (commented):
Hey! I did another round of review and found a few more things, sorry for the delay and back-and-forth 😞 Most of these can be punted but I think it'd be good to either address or discuss the issue with 404s/403s

jfdoming commented on src/app-openid.js:
```diff
@@ -0,0 +1,101 @@
+});
+
+app.get('/config', async (req, res) => {
+  const { cnt: ownerCount } = UserService.getOwnerCount() || {};
```
Sure! I'll make a ticket to track a fix for this. We should make sure to address this before we start advertising this feature, though.

lelemm commented on src/account-db.js:
```diff
@@ -38,74 +45,159 @@ export function getLoginMethod(req) {
-    return { error: 'invalid-password' };
+export async function disableOpenID(
+  loginSettings,
+  checkForOldPassword = false,
```
we used to have the possibility to enable openid from the config file. Since we changed the flow to need the user to bootstrap with password once, then enable openid, this was removed, but I forgot to remove this parameter

lelemm commented on src/services/user-service.js:
```diff
@@ -0,0 +1,273 @@
+  return owner;
+}
+
+export function getFileOwnerById(fileId) {
```
this method is not used anymore. removed

lelemm commented on src/load-config.js:
```diff
@@ -77,6 +77,8 @@ let defaultConfig = {
     fileSizeLimitMB: 20,
   },
   projectRoot,
+  multiuser: false,
```
changed to check openid enabled

lelemm commented on src/accounts/openid.js:
```diff
@@ -0,0 +1,319 @@
+      expiration =
+        Math.floor(Date.now() / 1000) + finalConfig.token_expiration * 60;
+    } else {
+      expiration = TOKEN_EXPIRATION_NEVER;
```
changed to 10 minutes fallback

lelemm commented on src/accounts/openid.js:
```diff
@@ -0,0 +1,319 @@
+  const code_challenge = generators.codeChallenge(code_verifier);
+
+  const now_time = Date.now();
+  const expiry_time = now_time + 300 * 1000;
```
ok

lelemm commented on src/app-admin.js:
```diff
@@ -0,0 +1,393 @@
+    return false;
+  }
+
+  const accesses = UserService.getUserAccess(
```
I changed the code to check if the user has access to this file, if not send it a forbidden message. This way a bad actor will not even have a response from the file not existing in first place

lelemm replied:
done

lelemm commented on src/app-admin.js:
```diff
@@ -0,0 +1,393 @@
+  };
+
+  if (granted === 0 && !isAdmin(session.user_id)) {
+    res.status(400).send({
```
done

lelemm commented on src/app-admin.js:
```diff
@@ -0,0 +1,393 @@
+  };
+
+  if (granted === 0 && !isAdmin(session.user_id)) {
+    res.status(400).send({
```
done

lelemm commented on src/app-admin.js:
```diff
@@ -0,0 +1,393 @@
+  };
+
+  if (granted === 0 && !isAdmin(res.locals.user_id)) {
+    res.status(400).send({
```
done

lelemm commented on src/app-admin.js:
```diff
@@ -0,0 +1,393 @@
+    };
+
+    if (granted === 0 && !isAdmin(res.locals.user_id)) {
+      res.status(400).send({
```
done

lelemm commented:
I changed all to 404 but add one missing forbidden on the get /access

lelemm commented on src/app-account.js:
```diff
@@ -26,22 +31,30 @@ export { app as handlers };
-  let { error, token } = bootstrap(req.body.password);
+app.post('/bootstrap', async (req, res) => {
+  let boot = await bootstrap(req.body);
 
-  if (error) {
-    res.status(400).send({ status: 'error', reason: error });
+  if (boot?.error) {
+    res.status(400).send({ status: 'error', reason: boot?.error });
     return;
   }
+  res.send({ status: 'ok', data: boot });
```
I don't want to leak errors

lelemm commented on src/app-admin.js:
```diff
@@ -0,0 +1,409 @@
+  };
+
+  if (granted === 0 && !isAdmin(session.user_id)) {
+    res.status(400).send({
```
the 404 is behind a forbidden 'wall', if the bad actor is a admin or a file owner, there is no leakage since they can already have access to file list

jfdoming reviewed (approved):
Not sure I quite understand why the /access endpoint doesn't leak info, but thanks for addressing the comments! Approving to unblock

lelemm commented:
don't approve this yet. found some inconsistencies

lelemm commented:
there was a begin transaction suggested by code rabbit hanging inside another transaction. now its all good

lelemm commented:
> Not sure I quite understand why the /access endpoint doesn't leak info, but thanks for addressing the comments! Approving to unblock

only admins or file owners can call this route.
- admins for all files
- file owners for their own files

jfdoming commented:
> > Not sure I quite understand why the /access endpoint doesn't leak info, but thanks for addressing the comments! Approving to unblock
> 
> only admins or file owners can call this route.
> 
> * admins for all files
> * file owners for their own files

Ah, I see—this one always returns 403 instead of always returning 404. Makes sense

jfdoming commented:
> there was a begin transaction suggested by code rabbit hanging inside another transaction. now its all good
Oof, that is unfortunate

lelemm commented:
> > there was a begin transaction suggested by code rabbit hanging inside another transaction. now its all good
> 
> Oof, that is unfortunate

Already fixed.
I'm working on the PR for this on the other repo

MatissJanis reviewed (approved):
Thanks!
